File: demo_module/app/src/main/java/com/bosh/module_demo_test/MainActivity.java
package com.bosh.module_demo_test;

import android.os.Bundle;

import com.alibaba.android.arouter.launcher.ARouter;

import androidx.appcompat.app.AppCompatActivity;

/**
 * @author bosh
 */
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
//        setContentView(R.layout.activity_main);
        ARouter.getInstance().build("/demo/demoMain").navigation();
        finish();
    }
}

File: kotlin_module/app/src/main/java/com/bosh/kotlin_module/MainActivity.kt
package com.bosh.kotlin_module

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView
import com.alibaba.android.arouter.launcher.ARouter
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
//        tv_test_main.setText("123")
        ARouter.getInstance().build("/kotlin/main").navigation()
        finish()
    }
}
